### PR TITLE
DebugVisualizer: equirect, removal/deconstruction, image matrix util, and unit test

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

This PR adds a set of features to the DebugVisualizer along with a basic unit test.

- Adds Equirectangular mode: `dbv.equirect=True` or `DebugVisualizer(sim, equirect=True)` to use an equirectangular sensor
- Adds `dbv.remove_dbv_agent()` - clean up after dbv by removing the added agent/sensor. This is now also called on deconstruction.
- Adds `dbv.clear_color` - set the background color of dbv images (must be set before initialization).
- Adds utility function `stitch_image_matrix` - stitches together a set of PIL Images into a single image matrix.

### Example Equirect Image from DBV:

Can be viewed in 3D with applications like https://arachnoid.com/3DViewer/
image

### Example stitched image matrix from a set of region omnidirectional renderings:

stitched_06_12_2024_081450952141-min

## How Has This Been Tested

Locally + CI tests

## Types of changes

- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes if required.

### CODE REVIEW

File: habitat-lab/habitat/sims/habitat_simulator/debug_visualizer.py [hunk 1]
```diff
@@ -539,15 +593,15 @@ def _peek_bb(
             world_transform = mn.Matrix4.identity_init()
         look_at = world_transform.transform_point(bb.center())
         bb_size = bb.size()
-        fov = self.sensor._spec.hfov
+        fov = 90 if self._equirect else self.sensor._spec.hfov
```
Comment (0mdc) [inline habitat-lab/habitat/sims/habitat_simulator/debug_visualizer.py:596]: In the 'normal' mode, `hfov` would be the default sensor field of view?

File: habitat-lab/habitat/sims/habitat_simulator/debug_visualizer.py
Comment (0mdc) [inline habitat-lab/habitat/sims/habitat_simulator/debug_visualizer.py:1]: Unrelated: You should fix the two `useage` typos in line `401` and `426`.

**Review by 0mdc — APPROVED** [review]
LGTM! Looks pretty useful. Could you share a screenshot of the output? :)

File: habitat-lab/habitat/sims/habitat_simulator/debug_visualizer.py (hunk 1 above)
Comment (aclegg3) [reply habitat-lab/habitat/sims/habitat_simulator/debug_visualizer.py:596]: The DebugVisualizer adds its own sensor, so whatever that sensor's fov is will be used here.

File: habitat-lab/habitat/sims/habitat_simulator/debug_visualizer.py
Comment (aclegg3) [reply habitat-lab/habitat/sims/habitat_simulator/debug_visualizer.py:1]: Haha, good catch. :monocle_face:

File: habitat-lab/habitat/sims/habitat_simulator/debug_visualizer.py [hunk 2]
```diff
@@ -17,9 +18,45 @@
...
+    bdo = DebugObservation(np.array(stitched_image))
+    bdo.image = stitched_image
+    return bdo
+
+
 class DebugObservation:
     """
     Observation wrapper to provide a simple interface for managing debug observations and caching the image.
+
+    NOTE: PIL.Image.Image.size is (width, height) while VisualSensor.resolution is (height, width)
```
Comment (0mdc) [inline habitat-lab/habitat/sims/habitat_simulator/debug_visualizer.py:59]: 👍

**Review by 0mdc — APPROVED** [review]
New changes look good!

Thanks for including images!

**Pull request merged**